### PR TITLE
created database using codefirst in entity framework

### PR DESCRIPTION
**Pull request by dggodi30**

### CODE REVIEW

File: Data/FitnessTrackerContext.cs [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+namespace FitnessTracker.Data
+{
+    public class FitnessTrackerContext : DbContext
+    {
+        public FitnessTrackerContext(DbContextOptions<FitnessTrackerContext> options) : base(options) { }
+
+        public DbSet<Log> Logs { get; set; }
+
+        public DbSet<Exercise> Exercises { get; set; }
+    }
```
Comment (athielking) [inline Data/FitnessTrackerContext.cs]: Look at the OnModelCreating override for the DB Context, you'll need to override and ensure your entities are configured within that so Foreign Keys and DB Constraints are set up properly.

File: Data/Entities/LogExercise.cs [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+using System.Threading.Tasks;
+
+namespace FitnessTracker.Data.Entities
+{
+    public class LogExercise
+    {
+        public int Id { get; set; }
+        public Log Log { get; set; }
+
+        public Exercise Exercise { get; set; }
```
Comment (athielking) [inline Data/Entities/LogExercise.cs]: I think you're going to need the foreign keys for Log and Exercise not just the entity reference

File: Data/Entities/Log.cs [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+using System.Threading.Tasks;
+
+namespace FitnessTracker.Data.Entities
+{
+    public class Log
+    {
+        public int Id { get; set; }
+
+        public User User { get; set; }
+
```
Comment (athielking) [inline Data/Entities/Log.cs]: Need actual foreign key for user

File: .config/dotnet-tools.json [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+  "isRoot": true,
+  "tools": {
+    "dotnet-ef": {
+      "version": "3.1.1",
+      "commands": [
+        "dotnet-ef"
+      ]
+    }
+  }
+}
```
Comment (athielking) [inline .config/dotnet-tools.json]: Please add a .gitignore file to ignore these system files like this and all of the files in the bin folder below which were added.  check here for some common .gitignore files https://github.com/github/gitignore

Comment (dggodi30) [reply .config/dotnet-tools.json]: I was thinking of adding a gitignore.  but I wasn't sure

Comment (athielking) [reply .config/dotnet-tools.json]: i think this file needs to be ignored still

File: Data/Entities/Log.cs (hunk 3 above)
Comment (dggodi30) [reply Data/Entities/Log.cs]: fixed by adding UserId as FK

File: Data/Entities/LogExercise.cs (hunk 2 above)
Comment (dggodi30) [reply Data/Entities/LogExercise.cs]: fixed by adding LogId and ExerciseId as FK's

File: Data/FitnessTrackerContext.cs (hunk 1 above)
Comment (dggodi30) [reply Data/FitnessTrackerContext.cs]: added OnModelingCreating

**Pull request merged**